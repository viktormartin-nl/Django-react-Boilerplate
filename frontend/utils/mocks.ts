import { Tip } from "../contexts/LandingPageContext";

export const tips: Tip[] = [
  { label: "Plan your application", key: "1" },
  { label: "Don't repeat yourself", key: "2" },
  { label: "Keep it simple, stupid", key: "3" },
  { label: "Control change", key: "4" },
  { label: "Document what you're doing", key: "5" },
  { label: "Test what you're writing", key: "6" }
];
