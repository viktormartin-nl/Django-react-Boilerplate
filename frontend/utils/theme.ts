import baseStyled, { ThemedStyledInterface } from "styled-components";

export const theme = {
  fontFamily: "font-family: 'Nunito Sans', sans-serif;",
  color: {
    white1: "#fff",
    white2: "rgba(255, 255, 255, 0.5)",
    white3: "#f7f7f7",
    blue1: "#0b3e71",
    blue2: "#1899d6",
    blue3: "#1cb0f6",
    blue4: "#235390",
    green1: "#78c800",
    green2: "#58a700",
    grey1: "#e5e5e5",
    grey2: "#4b4b4b"
  }
};

export type Theme = typeof theme;
export const styled = baseStyled as ThemedStyledInterface<Theme>;
