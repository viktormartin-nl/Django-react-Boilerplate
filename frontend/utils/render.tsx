import React from "react";
import { ThemeProvider } from "styled-components";
import { theme } from "./theme";
import { LandingPageProvider } from "../contexts/LandingPageContext";

export const withTheme = (children: React.ReactElement) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const withContextProvider = (children: React.ReactElement) => {
  return <LandingPageProvider>{children}</LandingPageProvider>;
};
