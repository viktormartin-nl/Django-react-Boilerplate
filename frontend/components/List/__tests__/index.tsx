import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { List } from "..";
import { tips } from "../../../utils/mocks";
import { fireEvent, render } from "@testing-library/react";
import { withTheme } from "../../../utils/render";

describe("<List /> test case", () => {
  test("test ids are in the document and items are rendered", () => {
    const { getByTestId, getAllByTestId } = render(
      withTheme(<List items={tips} />)
    );

    expect(getByTestId("unorderedList")).toBeInTheDocument();
    expect(getAllByTestId("listItem")).toHaveLength(tips.length);
  });

  test("renders items with extra functionalities", () => {
    const mockFunction = jest.fn();
    const { getAllByTestId } = render(
      withTheme(
        <List
          items={tips}
          removeItem={mockFunction}
          displayItemEditionModal={mockFunction}
        />
      )
    );

    expect(getAllByTestId("listItem")).toMatchSnapshot();
  });

  test("renders edit option when on edit mode", () => {
    const mockFunction = jest.fn();
    const { queryAllByText } = render(
      withTheme(
        <List
          items={tips}
          displayItemEditionModal={mockFunction}
          getCurrentTipLabel={mockFunction}
          setTipKey={mockFunction}
        />
      )
    );

    const editItemButtons = queryAllByText(/♻️/i);

    expect(editItemButtons).toHaveLength(tips.length);

    for (let i = 0; i < editItemButtons.length; i++) {
      fireEvent.click(editItemButtons[i]);
    }

    expect(mockFunction).toHaveBeenCalledTimes(editItemButtons.length * 3);
  });

  test("renders remove option when removeItem function is passed", () => {
    const removeItem = jest.fn();
    const { queryAllByText } = render(
      withTheme(<List items={tips} removeItem={removeItem} />)
    );

    const removeItemButtons = queryAllByText(/➖/i);
    expect(removeItemButtons).toHaveLength(tips.length);

    for (let i = 0; i < removeItemButtons.length; i++) {
      fireEvent.click(removeItemButtons[i]);
    }

    expect(removeItem).toHaveBeenCalledTimes(removeItemButtons.length);
  });
});
