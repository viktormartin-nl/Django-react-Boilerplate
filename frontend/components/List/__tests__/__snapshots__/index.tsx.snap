// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> test case renders items with extra functionalities 1`] = `
Array [
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-top: 2px solid #e5e5e5;
  line-height: 1.6;
  padding: 18px 22px;
  cursor: default;
}

.c1 {
  display: -webkit-box;
  display: -webkit-felx;
  display: -ms-felxbox;
  display: felx;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0px 8px;
}

.c1 span {
  cursor: pointer;
  margin: 0px 4px;
}

<li
    class="c0"
    data-testid="listItem"
    label="Plan your application"
  >
    Plan your application
    <div
      class="c1"
    >
      <span>
        ➖
      </span>
    </div>
  </li>,
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-top: 2px solid #e5e5e5;
  line-height: 1.6;
  padding: 18px 22px;
  cursor: default;
}

.c1 {
  display: -webkit-box;
  display: -webkit-felx;
  display: -ms-felxbox;
  display: felx;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0px 8px;
}

.c1 span {
  cursor: pointer;
  margin: 0px 4px;
}

<li
    class="c0"
    data-testid="listItem"
    label="Don't repeat yourself"
  >
    Don't repeat yourself
    <div
      class="c1"
    >
      <span>
        ➖
      </span>
    </div>
  </li>,
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-top: 2px solid #e5e5e5;
  line-height: 1.6;
  padding: 18px 22px;
  cursor: default;
}

.c1 {
  display: -webkit-box;
  display: -webkit-felx;
  display: -ms-felxbox;
  display: felx;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0px 8px;
}

.c1 span {
  cursor: pointer;
  margin: 0px 4px;
}

<li
    class="c0"
    data-testid="listItem"
    label="Keep it simple, stupid"
  >
    Keep it simple, stupid
    <div
      class="c1"
    >
      <span>
        ➖
      </span>
    </div>
  </li>,
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-top: 2px solid #e5e5e5;
  line-height: 1.6;
  padding: 18px 22px;
  cursor: default;
}

.c1 {
  display: -webkit-box;
  display: -webkit-felx;
  display: -ms-felxbox;
  display: felx;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0px 8px;
}

.c1 span {
  cursor: pointer;
  margin: 0px 4px;
}

<li
    class="c0"
    data-testid="listItem"
    label="Control change"
  >
    Control change
    <div
      class="c1"
    >
      <span>
        ➖
      </span>
    </div>
  </li>,
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-top: 2px solid #e5e5e5;
  line-height: 1.6;
  padding: 18px 22px;
  cursor: default;
}

.c1 {
  display: -webkit-box;
  display: -webkit-felx;
  display: -ms-felxbox;
  display: felx;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0px 8px;
}

.c1 span {
  cursor: pointer;
  margin: 0px 4px;
}

<li
    class="c0"
    data-testid="listItem"
    label="Document what you're doing"
  >
    Document what you're doing
    <div
      class="c1"
    >
      <span>
        ➖
      </span>
    </div>
  </li>,
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-top: 2px solid #e5e5e5;
  line-height: 1.6;
  padding: 18px 22px;
  cursor: default;
}

.c1 {
  display: -webkit-box;
  display: -webkit-felx;
  display: -ms-felxbox;
  display: felx;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0px 8px;
}

.c1 span {
  cursor: pointer;
  margin: 0px 4px;
}

<li
    class="c0"
    data-testid="listItem"
    label="Test what you're writing"
  >
    Test what you're writing
    <div
      class="c1"
    >
      <span>
        ➖
      </span>
    </div>
  </li>,
]
`;
